### PR TITLE
test(JestMigration): ScatterSpec.js -> Scatter.spec.tsx

### PR DESCRIPTION
**Pull request by proke03**
## Description
Migrate ScatterSpec.js -> Scatter.spec.tsx

## Related Issue
#3192

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: test-jest/cartesian/Scatter.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      <Surface width={500} height={500}>
+        <Scatter isAnimationActive={false} shape={renderCustomizedShape} points={data as any} />
+      </Surface>,
+    );
+
+    expect(container.querySelectorAll('.customized-shape')).toHaveLength(data.length);
+  });
+
+  test('Render customized line when line is set to be a ReactElement', () => {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars, react/prop-types
```
Comment (nikolasrieble) [inline test-jest/cartesian/Scatter.spec.tsx:61]: Is this required? Could you please try to not use it?

File: test-jest/cartesian/Scatter.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      <Surface width={500} height={500}>
+        <Scatter isAnimationActive={false} line={<CustomizedLine points={data as any} />} points={data as any} />
+      </Surface>,
+    );
+
+    expect(container.querySelectorAll('.customized-line')).toHaveLength(1);
+  });
+
+  test('Render customized line when line is set to be a function', () => {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (nikolasrieble) [inline test-jest/cartesian/Scatter.spec.tsx:73]: Same here.

File: test-jest/cartesian/Scatter.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    const renderCustomizedLine = ({ points }: { points: any }) => <path d="M0,0L200,200" className="customized-line" />;
```
Comment (nikolasrieble) [inline test-jest/cartesian/Scatter.spec.tsx:74]: Please generally do not use any if possible. You can infer the type automatically.

File: test-jest/cartesian/Scatter.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        <Scatter
+          isAnimationActive={false}
+          points={data as any}
+          onClick={onClick}
+          onMouseEnter={onMouseEnter}
+          onMouseLeave={onMouseLeave}
+        />
+      </Surface>,
+    );
+    const syb = container.querySelectorAll('.recharts-symbols')[4];
```
Comment (nikolasrieble) [inline test-jest/cartesian/Scatter.spec.tsx:100]: Please do not use abbreviations.
```suggestion
    const symbols = container.querySelectorAll('.recharts-symbols')[4];
```

File: test-jest/cartesian/Scatter.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    fireEvent.mouseEnter(syb);
+    expect(onMouseEnter).toHaveBeenCalled();
+    fireEvent.mouseLeave(syb);
+    expect(onMouseLeave).toHaveBeenCalled();
+    fireEvent.click(syb);
+    expect(onClick).toHaveBeenCalled();
```
Comment (nikolasrieble) [inline test-jest/cartesian/Scatter.spec.tsx:106]: Great test, I think we miss this test for many other components. This is a good example on how to test this feature.

**Review by nikolasrieble — APPROVED** [review]
Please do consider my suggestions. Nothing blocking though.

File: test-jest/cartesian/Scatter.spec.tsx (hunk 3 above)
Comment (proke03) [reply test-jest/cartesian/Scatter.spec.tsx]: I think it's ok in test code.

File: test-jest/cartesian/Scatter.spec.tsx (hunk 5 above)
Comment (proke03) [reply test-jest/cartesian/Scatter.spec.tsx:104]: Thank you!

File: test-jest/cartesian/Scatter.spec.tsx (hunk 3 above)
Comment (ckifer) [reply test-jest/cartesian/Scatter.spec.tsx]: Even in tests, `any` has a chance to make things happen that we don't expect. Unless there is an explicit reason (testing if something breaks with wrong props passed, there is no type in the code itself) we should stick to as strict types as we can. Maybe we can say if we use `any` we have to comment above it as to why we did

**Review by ckifer — APPROVED** [review]

**Pull request merged**